Improve the website design (again)

nisarhassan12 opened the pull request.

nisarhassan12 commented:
Most of the work is done apart from a few things:

- The new slider layout for Logos and tweets.
- The loading animation for the following:

![image](https://user-images.githubusercontent.com/46004116/83997552-11eaed80-a978-11ea-9c67-22ad1e3051d8.png)

- The orientation switcher for the tablet graphic:

![image](https://user-images.githubusercontent.com/46004116/83997643-4a8ac700-a978-11ea-82b2-9ac59d2397cd.png)

ChristinFrohne commented:
Thanks a lot!
Some early feedback:

- [ ] could you give the start screen a bit more space? A bit smaller screenshot, maybe a bit smaller font size for the headline. A bit more like this:
<img width="963" alt="Bildschirmfoto 2020-06-08 um 09 15 06" src="https://user-images.githubusercontent.com/46345125/84002360-99306500-a968-11ea-8a17-69e411f85a4c.png">

- [ ] the gitpod logo is btw still cut. Sven noticed it as well. We should keep on investigating in it 
<img width="234" alt="Bildschirmfoto 2020-06-08 um 09 17 08" src="https://user-images.githubusercontent.com/46345125/84002572-ef050d00-a968-11ea-9ee6-b5bb1ca9d6da.png">

- [ ] what do you think about having two button sizes? In the mockup the "log In" button is a bit smaller. By that we have a clear hierarchy. And you could use the same style for the "start workspace" button with the prefix 

- [ ] what font style are you currently using for the buttons? Bold? Could you try "semibold"?

- [ ] align the text so that `dev environments`is in one line and `no more tedious...`should be in a new line
<img width="617" alt="Bildschirmfoto 2020-06-08 um 08 50 38" src="https://user-images.githubusercontent.com/46345125/84001593-42765b80-a967-11ea-9ad7-08cbbcb83474.png">

- [ ] Currently the moving background doesn't look good. Could you try having it without the bounce? It could just slide in once without the bouncing and then stay there.  

- [ ] the background image should be behind the images (ipad, editor, all feature images)

- [ ] tweets should have a white background, maybe that's still coming with the sliding :)

- [ ] why did you delete the ice-cream-icon on "features"? Would be nice to still have it. 

- [ ] the square buttons could be a bit smaller (less spacing between icon and text)

- [ ] the border style should be consistent: I think having 1px everywhere is fine

- [ ] add hover style for `usual way`: white background -black font

- [ ] some boxes seem to have a border and some don't: please give all boxes a border
<img width="370" alt="Bildschirmfoto 2020-06-08 um 09 05 14" src="https://user-images.githubusercontent.com/46345125/84001579-38545d00-a967-11ea-9047-5787096b18d2.png">

- [ ] have the logos smaller
<img width="572" alt="Bildschirmfoto 2020-06-08 um 09 06 28" src="https://user-images.githubusercontent.com/46345125/84001674-689bfb80-a967-11ea-983c-ee74f388aae3.png">

- [ ] add a hover style for the contact icons in the footer

- [ ] use the same arrow style for moving up on "features" as for the upcoming slider
<img width="96" alt="Bildschirmfoto 2020-06-08 um 09 08 40" src="https://user-images.githubusercontent.com/46345125/84002805-4e631d00-a969-11ea-837b-06a45133a35b.png">

ChristinFrohne commented:
Tablet view:

- [ ] could you keep the boxes around the images and adjust the spacing. The text is very confusing to read currently
<img width="705" alt="Bildschirmfoto 2020-06-08 um 09 22 09" src="https://user-images.githubusercontent.com/46345125/84002984-9bdf8a00-a969-11ea-8329-96cce00df7ea.png">

- [ ] please add the same sliding look to the different examples as for the logos and tweets (only for tablet and mobile view)

- [ ] have the icon in the same color as the font color
<img width="259" alt="Bildschirmfoto 2020-06-08 um 09 25 08" src="https://user-images.githubusercontent.com/46345125/84003254-0395d500-a96a-11ea-93a7-a6a8d4f52207.png">

mobil:

- [ ] smaller image
<img width="492" alt="Bildschirmfoto 2020-06-08 um 09 26 48" src="https://user-images.githubusercontent.com/46345125/84003387-3e980880-a96a-11ea-8923-8b3c470b01fe.png">

- [ ] smaller tweets: one tweet should be complete visible
<img width="491" alt="Bildschirmfoto 2020-06-08 um 09 27 12" src="https://user-images.githubusercontent.com/46345125/84003410-49eb3400-a96a-11ea-982f-49cba1578e75.png">

- [ ] more spacing between the buttons
<img width="469" alt="Bildschirmfoto 2020-06-08 um 09 27 53" src="https://user-images.githubusercontent.com/46345125/84003464-5d969a80-a96a-11ea-9b64-2476960e8158.png">

nisarhassan12 commented:
> - [ ] what do you think about having two button sizes? In the mockup the "log In" button is a bit smaller. By that we have a clear hierarchy. And you could use the same style for the "start workspace" button with the prefix

I agree and that it will for sure make the hierarchy clearer.

nisarhassan12 commented:
> - [ ] what font style are you currently using for the buttons? Bold? Could you try "semibold"?

I'm currently using 600 as the font-weight for buttons. 400 hundred is used for normal text.

this is how 500 looks:

![image](https://user-images.githubusercontent.com/46004116/84138428-a8e2a300-aa67-11ea-95b8-f9d860dee6de.png)


looks very similar to 400.

nisarhassan12 commented:
> - [ ] why did you delete the ice-cream-icon on "features"? Would be nice to still have it.

@ChristinFrohne Thanks! It was not displayed in the mockup so I thought that you removed it. Sorry, I forgot that you old me that images are not displayed correctly in the mockup. I'll put it back.

nisarhassan12 commented:
> - [ ] the border style should be consistent: I think having 1px everywhere is fine

@ChristinFrohne Did you meant only the boxes/cards or buttons and other elements as well?

![image](https://user-images.githubusercontent.com/46004116/84147191-6aa0b000-aa76-11ea-823a-7c155bd7e874.png)

nisarhassan12 commented:
> - [ ] some boxes seem to have a border and some don't: please give all boxes a border

@ChristinFrohne the boxes don't have any border what you're seeing is a part of the image itself and there is no border on the right side in the image itself.

![image](https://user-images.githubusercontent.com/46004116/84148846-003d3f00-aa79-11ea-8ad4-d85ff64b3d57.png)

ChristinFrohne commented:
>Maybe it's just me I'm not too sure about the Ice Cream icon at the top of the features page i.e it's just some presentational stuff and it has no real semantic meanings so maybe we can remove it?

I like the ice-cream icon. The thought behind it is that you don't need it but everyone still loves having ice-cream. Similar to the features. They might not all be essential, but awesome to have. 
It also gives the page some color and happiness :)

We could move the icon a bit further up and make it a bit smaller so that the buttons to the features are all visible at first glance

nisarhassan12 commented:
Ok. Thanks.

ChristinFrohne commented:
>@ChristinFrohne Did you meant only the boxes/cards or buttons and other elements as well?
I only mean boxes and cards. The buttons are fine. Sorry for being not specific enough.

ChristinFrohne commented:
>@ChristinFrohne the boxes don't have any border what you're seeing is a part of the image itself and there is no border on the right side in the image itself.

That's weird, the image should have no borders. Did you use the file from the mockup (maybe something got mixed up there) or the one I sent you on slack?

nisarhassan12 commented:
I think I used the one from the Mockup.

nisarhassan12 commented:
The one you provided on Slack has the full opacity.

ChristinFrohne commented:
Ah okay. I thought you would manually adjust the opacity of the one I sent you? Shall I sent it to you again with 10% opacity?

nisarhassan12 commented:
I could technically adjust the opacity myself with code but I suspect that the size of the image will be smaller if we were to do this in Photoshop?

ChristinFrohne commented:
The size becomes a lot bigger if I save it as a .png. Would be great if you can adjust the file manually. Then it's also easier to adjust, in case the contrast is too low, and we want to change the opacity.

nisarhassan12 commented:
Ok. Thanks.

nisarhassan12 commented:
@ChristinFrohne I have updated the image and the border is now gone. this is how it looks now:

![image](https://user-images.githubusercontent.com/46004116/84157551-4a77ed80-aa84-11ea-81bd-da0ac1e0a75e.png)

nisarhassan12 commented:
> I only mean boxes and cards. The buttons are fine. Sorry for being not specific enough.

@ChristinFrohne Does this border on the left also has to be 1px? 

![image](https://user-images.githubusercontent.com/46004116/84165853-99765080-aa8d-11ea-9e19-985849cab4f0.png)

ChristinFrohne commented:
>@ChristinFrohne Does this border on the left also has to be 1px?
The line is fine like this. With 1px it would barely be visible

nisarhassan12 commented:
> - [ ] use the same arrow style for moving up on "features" as for the upcoming slider

@ChristinFrohne should I also make the colour for it grey?

ChristinFrohne commented:
>@ChristinFrohne Sorry! I'm unable to understand what examples you are talking about.

I'm talking about the example repos. Could you align them in the center and add the same sliding style as for the tweets and logos? (dots underneath, arrows on the side) This only applies to mobile view. 

<img width="508" alt="Bildschirmfoto 2020-06-11 um 08 07 12" src="https://user-images.githubusercontent.com/46345125/84350919-9331c280-abba-11ea-98ed-6db104b2db0f.png">

ChristinFrohne commented:
Great, it already looks so much better!

- [ ] you are still working on the logos, tweets and productivity bar, right?

- [ ] less spacing underneath the productivity circle

- [ ] let's not implement a new button style. Could you use the blue button instead?
<img width="744" alt="Bildschirmfoto 2020-06-11 um 08 11 50" src="https://user-images.githubusercontent.com/46345125/84351273-52867900-abbb-11ea-8311-7ae56071b969.png">

- [ ] I think the background behind the images should always move forward when you are in the middle of the site. But it should move backwards when it's not visible. What do you think? Maybe the movement could be smoother as well? We can also ask for others feedback

- [ ] less spacing after features and before `start for free`
<img width="1278" alt="Bildschirmfoto 2020-06-11 um 08 24 10" src="https://user-images.githubusercontent.com/46345125/84352082-f58bc280-abbc-11ea-91bf-4432eb68982c.png">
<img width="1219" alt="Bildschirmfoto 2020-06-11 um 08 24 54" src="https://user-images.githubusercontent.com/46345125/84352766-36380b80-abbe-11ea-8337-03d4a891f35c.png">


- [ ] background image of pricing -cloud and -self-hosted should be placed exactly the same.

- [ ] arrow of the cloud shouldn't be visible. Can you move it a bit or do you need another icon?
<img width="410" alt="Bildschirmfoto 2020-06-11 um 08 27 10" src="https://user-images.githubusercontent.com/46345125/84352307-6501b200-abbd-11ea-881a-019590ed796e.png">

- [ ] adjust icon size
<img width="191" alt="Bildschirmfoto 2020-06-11 um 08 29 56" src="https://user-images.githubusercontent.com/46345125/84352498-c164d180-abbd-11ea-9252-dd702f1958cf.png">

ChristinFrohne commented:
Some feedback for the careers page:

- [ ] can you have less spacing underneath `see job openings`

- [ ] could you shorten the paragraph a bit so that there is more spacing between the two columns?
<img width="1208" alt="Bildschirmfoto 2020-06-11 um 08 44 07" src="https://user-images.githubusercontent.com/46345125/84353614-caef3900-abbf-11ea-8c80-60c85eb79447.png">

- [ ] the team picture is not sharp. Please update it
![DSC_1182-small](https://user-images.githubusercontent.com/46345125/84353882-3a652880-abc0-11ea-8735-1cc2e2e18180.jpg)

- [ ] Could you color the active arrow in darker gray
<img width="509" alt="Bildschirmfoto 2020-06-11 um 08 48 57" src="https://user-images.githubusercontent.com/46345125/84354024-7a2c1000-abc0-11ea-9335-37d4f95ddb12.png">

- [ ] please use the button as shown in the mockup instead of text
<img width="559" alt="Bildschirmfoto 2020-06-11 um 08 50 42" src="https://user-images.githubusercontent.com/46345125/84354144-a9db1800-abc0-11ea-8a13-d9b02e5b68c8.png">

- [ ] the correct email address should be contact@typefox.io. Sorry for the confusion

nisarhassan12 commented:
>  - [ ] you are still working on the logos, tweets and productivity bar, right?

Thanks! I have started to work on the productivity bar i.e I'm learning stuff for that have not made any progress on the logos and tweets will try in the coming days.

nisarhassan12 commented:
> - [ ]  I think the background behind the images should always move forward when you are in the middle of the site. But it should move backwards when it's not visible. What do you think? Maybe the movement could be smoother as well? We can also ask for others feedback

personally, I don't like the moving back I think it's fine to have it not move back. Would be nice to ask others for the feedback. Do you have any example of a smoother animation over the web maybe we can take inspiration from that ?

nisarhassan12 commented:
![image](https://user-images.githubusercontent.com/46004116/84461567-33085280-ac86-11ea-92bc-9ac53579778c.png)

@ChristinFrohne Now the position stays the same but there is now a bit of extra space below the self-hosted pricing plans

 ![image](https://user-images.githubusercontent.com/46004116/84461505-0a805880-ac86-11ea-8439-26f51e8a93c8.png)

ChristinFrohne commented:
>personally, I don't like the moving back I think it's fine to have it not move back. Would be nice to ask others for the feedback. Do you have any example of a smoother animation over the web maybe we can take inspiration from that 

I'm also uncertain about. The idea was to have some movement & dynamic on the website. Other examples are https://github.com/features/codespaces or https://www.atlassian.com/. Feel free to try out something else if you get inspired.

nisarhassan12 commented:
As we have this white spacing on the bottom before the footer on careers page I have also added this to the blog.

![image](https://user-images.githubusercontent.com/46004116/84568158-fb3d0000-ad96-11ea-88de-c0ebe95c4e0f.png)

nisarhassan12 commented:
@ChristinFrohne for the edit in Gitpod button in Docs I will be in favour of the grey that we have instead of using white for the pen (to me the light grey looks easier on the eyes as compared to the white one). Whay do you say ? 

![image](https://user-images.githubusercontent.com/46004116/84569282-c3d25180-ad9e-11ea-8631-11844760e989.png)

![image](https://user-images.githubusercontent.com/46004116/84569297-e49aa700-ad9e-11ea-8c42-9ebbfc98d6e6.png)

nisarhassan12 commented:
@ChristinFrohne This seems to have a different border than the other UI elements. As you said earlier

![image](https://user-images.githubusercontent.com/46004116/84570231-82917000-ada5-11ea-8f87-2f5193b5585d.png)

Should I have the border style different for this element only? Thanks.

![image](https://user-images.githubusercontent.com/46004116/84570218-5a097600-ada5-11ea-83e3-e3c0779e24ef.png)

ChristinFrohne commented:
>Should I have the border style different for this element only? Thanks
It should be the same as the other ones: radius 3px, #E6E6E6

ChristinFrohne commented:
Thanks!
Some small comments:

- [ ] could you rearrange the background image on /blog? It doesn't look good if it's so widely stretched. It would be nice to have a search function but you could also work on that afterwards in a separated PR

<img width="961" alt="Bildschirmfoto 2020-06-15 um 14 35 41" src="https://user-images.githubusercontent.com/46345125/84658103-a673c380-af15-11ea-9fc2-018f4cea9fd4.png">

- [ ] could you use here the same arrow style as elsewhere? (circle with an inner arrow) 
<img width="727" alt="Bildschirmfoto 2020-06-15 um 14 44 29" src="https://user-images.githubusercontent.com/46345125/84658985-028b1780-af17-11ea-85f1-302bf10669d1.png">

- [ ] /docs navigation bar: it's not so nice to stretch the background image that wide. I repeated the image instead. Could you use this image?
![background-image-docs](https://user-images.githubusercontent.com/46345125/84659921-6bbf5a80-af18-11ea-8bf0-8e372b3e976e.jpg)


- [ ] The layout for each blog post still needs to be adjusted
<img width="962" alt="Bildschirmfoto 2020-06-15 um 14 36 17" src="https://user-images.githubusercontent.com/46345125/84658377-15e9b300-af16-11ea-84a0-5a0880a6c3c8.png">

- [ ] it doesn't seem like you're using a border radius of 3px everywhere. Any reason why not?
see for example index page, self-hosted, blog. Could you use the same style that you are using on /pricing everywhere?

- [ ] /features: When clicking on a button, could you make the page jump so that the correlating feature is displayed in the middle of the screen? Currently, it's aligned to the top which is confusing if multiple features are visible.

nisarhassan12 commented:
![image](https://user-images.githubusercontent.com/46004116/84679212-71359880-af4a-11ea-9d0a-756aa314c9ef.png)

Thanks, @ChristinFrohne indeed the stretched background does not look good other than that can you please outline what adjustments do you want me to make?

nisarhassan12 commented:
> - [ ]  it doesn't seem like you're using a border radius of 3px everywhere. Any reason why not?
see for example index page, self-hosted, blog. Could you use the same style that you are using on /pricing everywhere?

@ChristinFrohne I have added 3px border-radius to most of the elements. Please let me know if you find some element where it's still missing.

Also for a few components I have add only `boder-top-left-radius`, `border-top-right-radius` and vice versa. 

like for the post preview card image I think using a radial border for the bottom corners won't make that much sense. What do you say? Thanks.

![image](https://user-images.githubusercontent.com/46004116/84684800-a0e89e80-af52-11ea-836f-08a8af960b8d.png)

nisarhassan12 commented on src/components/index/Banner.tsx:
```diff
@@ -100,27 +95,23 @@ const Styled = styled.div`
                             <span>Start Instantly. Anywhere.</span>
                         </h1>
-                        <p>Gitpod launches ready-to-code dev environments<br /> for your GitHub or GitLab project with a single click.</p>
+                        <p>Gitpod launches automated dev environments for your <strong>GitHub, GitLab or Bitbucket</strong> project with a single click.</p>
```
I'm not sure whether the use of `<strong>` here is semantically correct.

According to [MDN](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/strong):

    

> The HTML Strong Importance Element `<strong>` indicates that its contents have strong importance, seriousness, or urgency. Browsers typically render the contents in bold type.

I don't feel like this phrase "GitHub, GitLab or Bitbucket" has any strong importance, seriousness or urgency? 
I'm a bit suspicious that it may sound weird to a Screen Reader users. Maybe I'm wrong. @jankeromnes What do you say? Thanks :slightly_smiling_face:

ChristinFrohne replied:
Search engine bots can recognize bold text, and this influences the ranking for the keywords. Important keywords should therefore be in headlines or/and highlighted in bold text. 
I also feel that bold elements break up the text and help to understand the product faster.

ChristinFrohne replied:
Could you keep the same order when listing the git hosting platforms? First GitLab, then GitHub and finally Bitbucket.
Thanks :)

nisarhassan12 commented on src/components/index/GetStarted.tsx:
```diff
@@ -0,0 +1,118 @@
+const GetStarted = () => (
+    <StyledGetStarted className="row">
+        <section className="get-started" id="get-started">
+            <h2><strong>Get Started</strong></h2>
```
Here too https://github.com/gitpod-io/website/pull/670/files#r440363682

ChristinFrohne commented:
>@ChristinFrohne Thanks! I have replaced the old image with the new repeated image but at width 1380px I had to make it stretch again because otherwise, this is how it would look (there is a white wedge to the left to the sidebar which does not look good):

I think the stretch is okay now. If you think differently, please let me know which size you need.

Currently, the image has a cut though. 
<img width="619" alt="Bildschirmfoto 2020-06-16 um 07 52 41" src="https://user-images.githubusercontent.com/46345125/84736586-6c9ecd80-afa6-11ea-8f52-3b6477f01aef.png">

nisarhassan12 commented:
> Currently, the image has a cut though.
> <img alt="Bildschirmfoto 2020-06-16 um 07 52 41" width="619" src="https://user-images.githubusercontent.com/46345125/84736586-6c9ecd80-afa6-11ea-8f52-3b6477f01aef.png">

@ChristinFrohne I have fixed this. Please have a look again whenever you'll have time Thanks.

nisarhassan12 commented on src/utils/features.tsx:
```diff
@@ -1,134 +1,312 @@
+        text: (<>
+            <p>Don’t postpone work because your current workspace is occupied with your work on the latest features.</p>
+            <p>To sneak in a bugfix on a maintenance branch or a code review - just start more workspaces - in parallel.</p>
+            <p><strong>Use clean, fresh workspaces for every task.</strong></p>
```
Thanks @ChristinFrohneI have removed this `Use clean, fresh workspaces for every task.`

nisarhassan12 commented on src/utils/features.tsx:
```diff
@@ -1,134 +1,312 @@
+    },
+    {
+        id: 'parallel',
+        src: ParallelWorkspaces,
```
Sorry! I didn't noticed I thought it was the same old image but now I have updated it Thanks.

nisarhassan12 commented on src/utils/features.tsx:
```diff
@@ -1,134 +1,312 @@
+        alt: 'Supported Programming Languages',
+        title: <>Benefit from<br />Industry-Leading<br /><strong>Code Intelligence</strong></>,
+        text: (<>
+            <p>Gitpod supports 20+ programming languages out of the box which are continuously improved. If your favourite language is missing: simply find your VS Code language extension via the vendor neutral marketplace of <a href="https://open-vsx.org/" target="_blank">VSX Registry</a> and drag & drop it into Gitpod.</p>
```
What do you say about the following:


> Gitpod supports 20+ programming languages out of the box and is continuously extended.
>
> If your favourite language is missing: simply drag & drop your VS Code language extension into Gitpod or install it via the integrated Open VSX marketplace.

nisarhassan12 commented:
> - [ ] the gitpod logo is btw still cut. Sven noticed it as well. We should keep on investigating in it

![image](https://user-images.githubusercontent.com/46004116/84917817-40379e00-b0d9-11ea-89e5-b19b872f8ade.png)

I also noticed it on Chrome. I'm not sure what is the cause for this but increasing the height of the logo from `4rem` to `4.2rem` seems to fix the problem. Also to neutralize the effect so that the logo does not look big I have scaled it down to `.97`.

We are having `object-fit: contain` for all the images globally and maybe there is a problem with the `ViewBox` for the logo not sure.

@ChristinFrohne Please have a look whenever you'll have time I hope the issue will be fixed now.

nisarhassan12 commented:
@ChristinFrohne I have added the back to top button to the blog and the documentation would it make sense to add it to the other pages as well? like `careers`, `self-hosted` and so on and so forth?

nisarhassan12 commented:
@ChristinFrohne The design for `imprint`, `terms` and `privacy` policy is not updated. Is the old design ok for these pages ?

nisarhassan12 commented:
@ChristinFrohne only for the buttons we're downloading the `500` weight of `Montserrat` now we are downloading 3 weights `400, 500, 600`

This is how `500` for the button looks:

![image](https://user-images.githubusercontent.com/46004116/84978120-8de9f000-b145-11ea-9b94-5768882b4632.png)

This is how `400` looks:

![image](https://user-images.githubusercontent.com/46004116/84978152-9e01cf80-b145-11ea-84f1-a29e68f29540.png)

Can we use `400` instead of `500` or maybe revert back to `600` for the buttons that way we only have to download 2 font weights instead of 3 which is better for performance ?

ChristinFrohne commented:
>@ChristinFrohne I have added the back to top button to the blog and the documentation would it make sense to add it to the other pages as well? like careers, self-hosted and so on and so forth?

I think, it's fine like this. The other pages are not that long. Could you make sure that all the icon-buttons have the same style ("back-to-top-button", "edit-docs-button", "slide-through-logos/tweets-button", "next-documentation-page", "job-opening-button" etc.) I'd say having all icon-buttons light gray by default and dark gray on hover, would be good. Just like the icon-buttons on /careers. Their size also seems good, the buttons shouldn't be any bigger rather smaller.

ChristinFrohne commented:
>@ChristinFrohne The design for imprint, terms and privacy policy is not updated. Is the old design ok for these pages ?

Good point, we should use everywhere the same style. I just added an example for it to the mockup link. Please also remember to implement the changed layout for each blog post.

ChristinFrohne commented:
>@ChristinFrohne only for the buttons we're downloading the 500 weight of Montserrat now we are downloading 3 weights 400, 500, 600

I agree we should keep it simple. Please use a font-weight of 400 for all buttons

jankeromnes commented on src/components/index/OpenSource.tsx:
```diff
@@ -1,100 +1,44 @@
+            <div className="text">
+                <img src={OpenSourceImg} alt=""/>
+                <h2><strong>Giving Back to Open-Source!</strong></h2>
+                <p>Gitpod is built on open-source and wouldn’t exist without it. We’re happy to give something back by supporting the open-source community with a <Link to="/docs/professional-open-source/">free Open-Source Plan</Link>.</p>
```
We're not really giving out free plans to everyone. We're sponsoring professional open-source developers with unlimited hours.

```suggestion
                <p>Gitpod is built on open-source and wouldn’t exist without it. We’re happy to give something back by sponsoring <Link to="/docs/professional-open-source/">Professional Open-Source</Link> developers with unlimited Gitpod hours on open-source repositories.</p>
```

ChristinFrohne replied:
This is referring to the free 50hours open-source plan and not to the professional open-source plan. The professional plan only affects a few users while the free open-source plan is relevant for everyone. That's why I thought it makes more sense to mention it. 

Maybe instead we could mention the 50hours/ month limitation.

jankeromnes replied:
> This is referring to the free 50hours open-source plan and not to the professional open-source plan.

Ah, in this case maybe:

```suggestion
                <p>Gitpod is built on open-source and wouldn’t exist without it. We’re happy to give something back by being completely <strong>free for open source</strong>.</p>
```

jankeromnes commented on src/components/MoreInfo.tsx:
```diff
@@ -0,0 +1,107 @@
+        Img = <img src={MapGrey} alt="Explore Gitpod" />
+        Title = <><strong>Explore</strong> Gitpod</>
+        Text = (<>
+            Learn about collaboration, shared workspace and snapshots, supported programming languages, and much more.
```
```suggestion
            Learn about collaboration, workspace sharing, reproducible snapshots, supported language features, and much more.
```

ChristinFrohne commented:
Thanks a lot for your review @jankeromnes :) 

> The left/right social proof buttons look distracting (more visible than logos). Also clicking them doesn't work.
> The three big dots below are confusing and pointless. Please remove them.

Please note that this section is still WIP. The arrow icons are supposed to only appear on hover. the dots underneath visualize that there are three pages of icons. The first one should be darker

> There is waaaaay too much vertical whitespace here. This distances to scroll are far too long. I'm not a marathon scroller.

True, that because we recently deleted the productivity bar. What do you think about something like this instead? (needs another icon than the professional plan) Aim: stating Gitpod's value
<img width="959" alt="Bildschirmfoto 2020-06-18 um 15 16 31" src="https://user-images.githubusercontent.com/46345125/85025298-82172180-b177-11ea-8056-3f0bc4eeeca7.png">

>The tablet looks a bit ugly. As a developer, I don't want to code on a cheap / nameless / off-brand tablet. I want to code on an iPad. Please use a nicer-looking picture (ideally something that looks like an iPad).

:D sure, I can see about an alternative

> That image also looks a bit ugly (inconsistent spacing, inconsistent font size, and nobody uses Times New Roman in such images anymore) -- I wouldn't want to spend all my days working in this interface. Please use a nicer image, with consistent spacing and font-sizes (and no serifs).

Interesting, for me it's shown in Montserrat
<img width="1196" alt="Bildschirmfoto 2020-06-18 um 15 20 50" src="https://user-images.githubusercontent.com/46345125/85025168-5005bf80-b177-11ea-833f-be41810d3669.png">

> Too much vertical whitespace. Please remove some. (related to tweets)

This section is also work in progress. It's supposed to get the same style as the logos above: aligned to center and with arrows to click through

jankeromnes commented:
No worries! (Reviewing is the easy part, you both did all the hard work already and the PR looks great already.)

> The arrow icons are supposed to only appear on hover.

Ideally the social proof logos should still be horizontally scrollable. Requiring (many) clicks or taps to reveal UI elements is not good UX.

> the dots underneath visualize that there are three pages of icons. The first one should be darker

I still think the dots are a bit pointless (you can already use the arrows), i.e. they're extra work + extra visual noise + not really necessary.

> What do you think about something like this instead? (needs another icon than the professional plan) Aim: stating Gitpod's value

That looks good to me, but it's:
- "of watching your code build" and not "of watching your code to build"
- "you can prebuild" and not "you can prebuilt" (also, chronologically, you "automate" first and "prebuild" afterwards, so "you can automate and prebuild")

jankeromnes commented:
> > The tablet looks a bit ugly. As a developer, I don't want to code on a cheap / nameless / off-brand tablet. I want to code on an iPad. Please use a nicer-looking picture (ideally something that looks like an iPad).
> 
> :D sure, I can see about an alternative

Ah! Also, the tablet should probably be in landscape mode. I don't think anyone codes in landscape mode. See also:
- https://twitter.com/d6y/status/1219305442226118658
- https://twitter.com/katee_kt/status/1204430283556016129
- https://twitter.com/edbrsk/status/1187455106318557186
- https://twitter.com/DmytroGladkyi/status/1185265163529834498

ChristinFrohne commented:
>I much prefer the horizontally scrollable tweets as they are today. On the web we're not used to clickable buttons when we want to see more content, we're used to scrolling (especially on mobile).

Currently, you don't see that you can scroll vertically. It doesn't mean that the scrolling needs to be disabled. We can add the arrows additionally for users without a touchpad and to visual the option of sliding through. 
<img width="1359" alt="Bildschirmfoto 2020-06-19 um 09 03 38" src="https://user-images.githubusercontent.com/46345125/85106544-a8d36780-b20c-11ea-9b4d-175b9dfd061c.png">

Panchajanya1999 commented:
Looks Good :)

nisarhassan12 commented:
> Good point, we should use everywhere the same style. I just added an example for it to the mockup link. Please also remember to implement the changed layout for each blog post.

Thanks! I have updated the layout. This is how the table on the privacy page looks should we add a different background to it or is it ok like what we already have:

![image](https://user-images.githubusercontent.com/46004116/85164623-110e6180-b27e-11ea-87fb-be2fc4905ba2.png)

ChristinFrohne commented:
>Thanks! I have updated the layout. This is how the table on the privacy page looks should we add a different background to it or is it ok like what we already have

It's okay for now. Let's rather focus on the other remaining issues :)

nisarhassan12 commented:
> - [ ] There is waaaaay too much vertical whitespace here. This distances to scroll are far too long. I'm not a marathon scroller.

Thanks I have reduced the spacing around the features but not between them. @ChristinFrohne can you have a look and tell if that looks ok now? Thanks.

nisarhassan12 commented:
> Currently, you don't see that you can scroll vertically. It doesn't mean that the scrolling needs to be disabled. We can add the arrows additionally for users without a touchpad and to visual the option of sliding through.

Thanks, @ChristinFrohne this problem can be solved we can have blurry edges for the container and set the max-width of the child to 90% as an example see https://reactjs.org

![image](https://user-images.githubusercontent.com/46004116/85281250-1698c100-b4a3-11ea-9c5b-b13f23cdcc53.png)

ChristinFrohne commented:
>Thanks I have reduced the spacing around the features but not between them. @ChristinFrohne can you have a look and tell if that looks ok now? Thanks

Could you reduce the spacing before the full automation-section a bit? Then it's good :)

ChristinFrohne commented:
Great thanks. 
Some small nits:

- [ ] please add a button linking to "contribute.dev" from the open-source section

- [ ] have less spacing above the icon in the security box, add a break before "Hardened by professional ..."

- [ ] align "trusted by 200,000 developers" in the center

- [ ] oh you decided not to use the drop down option for "start workspaces"? If you'd like to keep it this way, please make the GitHub logo black and use the usual hover style (dark gray background)

- [ ] "start workspace" button looks inconsistent. It has a different height than the "log in"-button. Please, use the same style as for the "log in"-button. The length can differ

- [ ] add a 3px radius to the vs code extension image

- [ ] add more spacing to docs 
<img width="1297" alt="Bildschirmfoto 2020-06-24 um 11 15 29" src="https://user-images.githubusercontent.com/46345125/85528397-1a942280-b60c-11ea-9a9f-5f299d260ef5.png">

- [ ] please always use the same style for the circle-buttons. Currently, they all differ in size. Make them a bit smaller, let's say 40x40px and no white border

- [ ] please remove the shadow for the blog posts, add border, adjust blog headlines as shown in the mockup 
<img width="635" alt="Bildschirmfoto 2020-06-24 um 11 20 15" src="https://user-images.githubusercontent.com/46345125/85528823-b32aa280-b60c-11ea-8f02-a9e13edee43d.png">

- [ ] /careers could you add the hover effect to the circle-buttons

- [ ] /about exchange "explore Gitpod"-section with current design

- [ ] remove sliding background effect

ChristinFrohne commented on src/utils/index.tsx:
```diff
@@ -0,0 +1,41 @@
+        alt: 'GitHub, GitLab and Bitbucket',
+        title: <>Stay in the Flow -<br /><strong>Integrated with GitLab, GitHub, Bitbucket</strong></>,
+        text: (<>
+            <p>Start your workflow from your Git hosting platform and let Gitpod beam you in a ready-to-code dev environment.</p>
```
Can we shorten this? Maybe just:
"Whether you want to fix a bug, add a feature or do a code review, Gitpod beams you in a ready-to-code dev environment.

ChristinFrohne commented on src/pages/index.tsx:
```diff
@@ -113,49 +113,54 @@ const IndexPage: React.SFC<{}> = () => (
+            alt="Open Source"
+            title="Free For Open-Source!"
+            text={<>Gitpod is built on open-source and wouldn’t exist without it. We’re happy to give something back by being completely <strong>free for open source</strong>.</>}
         />
```
Add button linking to contribute.dev

nisarhassan12 commented on src/pages/index.tsx:
```diff
@@ -157,7 +157,9 @@ const IndexPage: React.SFC<{}> = () => (
             title="Free For Open-Source!"
-            text={<>Gitpod is built on open-source and wouldn’t exist without it. We’re happy to give something back by being completely <strong>free for open source</strong>.</>}
+            text={<> <p> Gitpod is built on open-source and wouldn’t exist without it. We’re happy to give something back by being completely <strong>free for open source</strong>. </p>
+            <p>Find your next project on contribute.dev. </p>
```
@ChristinFrohne This paragraphs without context looks a bit odd.

ChristinFrohne replied:
What about adding something like this? "Find your next project on contribute.dev, a platform listing open-source projects with a ready-to-code setup." Maybe we could replace "project" with another word to not have it twice in there

nisarhassan12 replied:
@jankeromnes What do you think :slightly_smiling_face:? Thanks

nisarhassan12 commented:
> - [ ] oh you decided not to use the drop down option for "start workspaces"? If you'd like to keep it this way, please make the GitHub logo black and use the usual hover style (dark gray background)

Thanks @ChristinFrohne I'm still in the favour of a drop-down but there was an issue with the implementation on small devices where there is horizontal scrolling so I gave up and decided to fall back to this implementation.

nisarhassan12 commented:
> - [x]  oh you decided not to use the drop down option for "start workspaces"? If you'd like to keep it this way, please make the GitHub logo black and use the usual hover style (dark gray background)

@ChristinFrohne This is how it looks while having the usual hover style. Should I revert back to what I had previously? 

![image](https://user-images.githubusercontent.com/46004116/85562336-d291f180-b645-11ea-8f50-9df403f658ba.png)

nisarhassan12 commented:
The scroll animation that we have added does not work well for keyboard users see the below gif. The problem is that we are having a scroll listener on each of the features which when the `getClientBoundingRect()` satisfies the following:
```js
rect.top >= -100 && rect.bottom <= window.innerHeight + 100
```
does the animation but when the user tabs `scroll` event does get fired but the condition does not get satisfied most of the times and the contents do not show up the links and so on are still tabbable but the opacity for the contents never becomes `1`.

![Peek 2020-06-30 13-58](https://user-images.githubusercontent.com/46004116/86106390-f0de6e00-bad9-11ea-8e4e-5c15aa34f103.gif)

jankeromnes commented on src/utils/index.tsx:
```diff
@@ -0,0 +1,41 @@
+    {
+        id: 'automation',
+        Graphic: FullAutomationGraphics,
+        title: <>No more Waiting with<br /><strong>Full Automation</strong></>,
```
```suggestion
        title: <>Skip the Waiting with<br /><strong>Full Automation</strong></>,
```

jankeromnes commented:
Note: I have only reviewed the front page. I still need to review the Features page, and all the other pages that were also modified.

JesterOrNot commented on src/pages/careers.tsx:
```diff
@@ -0,0 +1,210 @@
+const features: FeatureItemProps[] = [
+    {
+        title: "Flat Hierarchy",
+        text: "We don’t like hierarchies. At Gitpod everyones opinion counts.",
```
"We don’t like hierarchies. At Gitpod everyones opinion counts." --> "We don’t like hierarchies. At Gitpod everyone's opinion counts."

JesterOrNot commented on src/pages/careers.tsx:
```diff
@@ -0,0 +1,210 @@
+    },
+    {
+        title: "People Who Care",
+        text: "We support each other and grow to new strength together. On regular team events we’re empowering this bond.",
```
"We support each other and grow to new strength together. On regular team events we’re empowering this bond." --> "We support each other and grow to new strengths together. On regular team events, we’re empowering this bond."

JesterOrNot commented on src/pages/careers.tsx:
```diff
@@ -0,0 +1,210 @@
+    },
+    {
+        title: "Worthy Mission",
+        text: "Together we want to make developers life easier by providing the best software tools.",
```
"Together we want to make developers life easier by providing the best software tools." --> "Together we want to make developer's lives easier by providing the best software tools."

JesterOrNot commented on src/pages/careers.tsx:
```diff
@@ -0,0 +1,210 @@
+    {
+        title: 'Developer Tools Engineer',
+        intro: 'If you were ever frustrated with a development tool you used and decided to build a better one yourself, this offer is for you.',
+        paragraphs: `We at TypeFox specialize in building tools and we are looking to expand our team. Our mission is to make software development fun and remove all tedious friction. We are a highly motivated team, organized in a flat hierarchy, and led by technical founders. We love open source, have started and lead many popular open-source projects, and contribute to many others. In this fash ...
```
"We at TypeFox specialize in building tools and we are looking to expand our team. Our mission is to make software development fun and remove all tedious friction. We are a highly motivated team, organized in a flat hierarchy, and led by technical founders. We love open source, have started and lead many popular open-source projects, and contribute to many others. In this fashion, we present at international conferences and collaborate openly around the globe." --> "We at TypeFox specialize in building tools and we are looking to expand our team. Our mission is to make software development fun and remove all tedious friction. We are a highly motivated team, organized in a flat hierarchy, and led by technical founders. We love open-source, have started and lead many popular open-source projects, and contribute to many others. In this fashion, we present at international conferences and collaborate openly around the globe."

JesterOrNot commented on src/pages/careers.tsx:
```diff
@@ -0,0 +1,210 @@
+            },
+            {
+                title: 'We offer:',
+                items: ['Space for creativity - feel free to create the non-existing.', 'Collaborate with a highly motivated team with a strong technical background', 'Flexible working hours.', 'Work remotely from Germany or France, or from the office in Kiel (Germany).']
```
['Space for creativity - feel free to create the non-existing.', 'Collaborate with a highly motivated team with a strong technical background', 'Flexible working hours.', 'Work remotely from Germany or France, or from the office in Kiel (Germany).'] --> ['Space for creativity - feel free to create the non-existing.', 'Collaborate with a highly motivated team with a strong technical background', 'Flexible working hours.', 'Work remotely from Germany or France, or the office in Kiel (Germany).']

JesterOrNot commented on src/pages/careers.tsx:
```diff
@@ -0,0 +1,210 @@
+        lists: [
+            {
+                title: 'What you will do:',
+                items: ['Build desktop and cloud-based tools using Go, TypeScript and Java.', 'Create, use and dog-food cutting-edge technology.', 'Work with communities, users and clients.']
```
['Build desktop and cloud-based tools using Go, TypeScript and Java.', 'Create, use and dog-food cutting-edge technology.', 'Work with communities, users and clients.'] --> ['Build desktop and cloud-based tools using Go, TypeScript, and Java.', 'Create, use, and dog-food cutting-edge technology.', 'Work with communities, users, and clients.']

What does `Create, use and dog-food cutting-edge technology.` even mean?

nisarhassan12 replied:
> What does `Create, use and dog-food cutting-edge technology.` even mean?

Sorry, I don't know @jankeromnes can you please explain this?

JesterOrNot commented on src/pages/careers.tsx:
```diff
@@ -0,0 +1,210 @@
+    {
+        title: 'UX Developer',
+        intro: 'Gitpod provides ready-to-code development environments for professional software engineering teams. We set out to change how developers work by eliminating tedious and repetitive configuration of local machines and replacing it with a fully integrated, cloud-based solution. Gitpod is the missing piece in a modern DevOps toolchain.',
+        paragraphs: `As a UX Developer, you will take care of the UI/UX design and development of Gitpod. You will become an essential part of a cross-functional team and directly work together with product management, development and marketing.`,
```
"As a UX Developer, you will take care of the UI/UX design and development of Gitpod. You will become an essential part of a cross-functional team and directly work together with product management, development and marketing." --> "`As a UX Developer, you will take care of the UI/UX design and development of Gitpod. You will become an essential part of a cross-functional team and directly work together with product management, development, and marketing.`"

JesterOrNot commented on src/pages/careers.tsx:
```diff
@@ -0,0 +1,210 @@
+        lists: [
+            {
+                title: 'What you will do:',
+                items: ['Design, prototype, and iterate on the UX of new features and refresh existing UX.', 'Communicate design solutions through wireframes, mockups, and interactive prototypes.', 'Implement mockups using HTML and TypeScript/JavaScript.', 'Work closely with engineering and be the "go to" person when user interface designs require clarification or change.']
```
['Design, prototype, and iterate on the UX of new features and refresh existing UX.', 'Communicate design solutions through wireframes, mockups, and interactive prototypes.', 'Implement mockups using HTML and TypeScript/JavaScript.', 'Work closely with engineering and be the "go to" person when user interface designs require clarification or change.'] --> ['Design, prototype, and iterate on the UX of new features and refresh existing UX.', 'Communicate design solutions through wireframes, mockups, and interactive prototypes.', 'Implement mockups using HTML and TypeScript/JavaScript.', 'Work closely with engineering and be the "go-to" person when user interface designs require clarification or change.']

JesterOrNot commented on src/pages/careers.tsx:
```diff
@@ -0,0 +1,210 @@
+            },
+            {
+                title: 'What we are looking for:',
+                items: ['3+ years experience designing UI / UX and web development (preferably in a B2B SaaS space).', 'Solid knowledge of developer workflows and tools.', 'A team player who can also work independently.', 'A curious mindset with a drive to learn and grow new skills.']
```
['3+ years experience designing UI / UX and web development (preferably in a B2B SaaS space).', 'Solid knowledge of developer workflows and tools.', 'A team player who can also work independently.', 'A curious mindset with a drive to learn and grow new skills.'] --> ['3+ years of experience designing UI / UX and web development (preferably in a B2B SaaS space).', 'Solid knowledge of developer workflows and tools.', 'A team player who can also work independently.', 'A curious mindset with a drive to learn and grow new skills.']

JesterOrNot commented on src/pages/careers.tsx:
```diff
@@ -0,0 +1,210 @@
+            },
+            {
+                title: 'Your benefits:',
+                items: ['Space for creativity - feel free to create the non-existing.', 'Collaborate with a highly motivated team with a strong technical background', 'Flexible working hours.', 'Work remotely from Germany or France, or from the office in Kiel (Germany).']
```
['Space for creativity - feel free to create the non-existing.', 'Collaborate with a highly motivated team with a strong technical background', 'Flexible working hours.', 'Work remotely from Germany or France, or from the office in Kiel (Germany).'] --> ['Space for creativity - feel free to create the non-existing.', 'Collaborate with a highly motivated team with a strong technical background', 'Flexible working hours.', 'Work remotely from Germany or France, or the office in Kiel (Germany).']

nisarhassan12 commented:
Many Thanks, @jankeromnes for the review. 

> - [ ] The rocket picture looks pixelated & not sharp. Please don't stretch images, but let them take their natural size.

the original image size is `276x360` currently we've downscaled it i.e the size on the page is `190.333x248.267`:

![image](https://user-images.githubusercontent.com/46004116/86236650-ded0fe00-bbb3-11ea-91b3-b1fffab414bb.png)

The problem is not the alteration of size rather it's the actual image itself.

nisarhassan12 commented:
> - [ ] Hovering over an example link doesn't show the URL or a tooltip.

Thanks, @jankeromnes. I have added a `title="Choose your platform"` on the buttons this shows up the built-in browser tool-tip on hover and I think it does the job what do you say or should I add a custom tooltip? 

> - [ ] I have no idea what it is that I am clicking on, so I'll better not risk it. Please make these things actual <a/> links, so that when you hover over them, you can see the URL in the bottom left.

I'm sorry if I misunderstood you the button when clicked goes on to show up the 3 actual links(where on hover the user can see the URL) to the supported platforms. I think you said this because there was nothing showing up on hover which is fixed by adding a title to the `Start Workspace` button. Is that correct or did you meant something else? Thanks

![image](https://user-images.githubusercontent.com/46004116/86237657-9ca8bc00-bbb5-11ea-8f2c-106b8039a4b6.png)

jankeromnes commented on src/utils/index.tsx:
```diff
@@ -0,0 +1,41 @@
+        alt: 'GitHub, GitLab and Bitbucket',
+        title: <>Stay in the Flow -<br /><strong>Integrated with GitLab, GitHub, Bitbucket</strong></>,
+        text: (<>
+            <p>Start your workflow from your Git hosting platform and let Gitpod beam you in a ready-to-code dev environment.</p>
```
```suggestion
            <p>Start your workflow from your Git hosting platform and let Gitpod beam you in a prebuilt dev environment.</p>
```

nisarhassan12 replied:
@jankeromnes Thanks. Maybe it's just me I'm not too sure to me "ready-to-code" looks more descriptive and I feel like`prebuilt` is a lesser-known word. What do you say?

jankeromnes commented on src/utils/projects.tsx:
```diff
@@ -9,13 +9,13 @@ export const projects = [
         image: JAVA,
         title: 'Java & Spring',
-        href: "https://github.com/gitpod-io/spring-petclinic",
+        href: "gitpod-io/spring-petclinic",
```
Ah, now I see, the "Start Workspace" button gives you a choice between 3 platform.

However, the `gitpod-io/spring-petclinic` part only works for `github.com` (because on `gitlab.com` and on `bitbucket.org`, the organization is called `gitpod` and not `gitpod-io`).

Maybe, instead of a common URI, you could have 3 full URLs? E.g.

```suggestion
        githubUrl: "https://github.com/gitpod-io/spring-petclinic",
        gitlabUrl: "https://gitlab.com/gitpod/spring-petclinic",
        bitbucketUrl: "https://bitbucket.org/gitpod/spring-petclinic",
```

The other advantage of doing this is that, if an example is not present on one platform, you just don't specify that platform's URL, and then in the React component you can simply hide platforms that have no URLs.

jankeromnes commented on src/components/index/GetStarted.tsx:
```diff
@@ -0,0 +1,118 @@
+            <p>Free for Open Source, and free for 30 days with private repositories.</p>
+
+            <div className="get-started__prefix">
+                <div className="get-started__prefix-text-container">
+                    <h3 className="get-started__prefix-text">Prefix any GitHub or GitLab<br aria-hidden={true} />URL with <strong>gitpod.io/#</strong></h3>
+                </div>
```
In https://github.com/gitpod-io/website/pull/670/files/b2ead4e6706cb580c66cea2bb2b3a1dbeca5bf9c#r447656300 I suggest explaining the prefix above the widget.

This means that this left part of the widget is no longer necessary. Maybe let's remove it, and have the widget use the full width:

```suggestion
```

nisarhassan12 commented:
> - [ ] The image looks pixelated & not sharp. Please don't stretch raster images but let them take their natural size.

@jankeromnes Thanks it's not stretched rather it's size is scaled down see https://github.com/gitpod-io/website/pull/670#issuecomment-652354692.

nisarhassan12 commented:
> In general, it feels very random that all the text for our pages is implemented in many separate components. Instead of having one page with all the content in it, we now have a ton of code that references other code and it's very hard to find what I'm looking for.
>
> In the future, please only make separate components when they're actually generic & re-usable. There is no need to make separate components for one feature that only shows up once in the website (in that case, it's better to in-line the HTML code directly into the page component).

Thanks @jankeromnes I will try to keep this in mind.

nisarhassan12 commented on src/utils/index.tsx:
```diff
@@ -0,0 +1,41 @@
+        Graphic: FullAutomationGraphics,
+        title: <>No more Waiting with<br /><strong>Full Automation</strong></>,
+        text: (<>
+            <p>Skip the friction and get beamed into a fully setup, compiled and tested dev environment with one click.</p>
```
Thanks! I have done the double-checking. Thanks

nisarhassan12 commented:
> The URL prefix widget is very confusing now. It looks so much like a web browser that I now think it's just an image. I have no idea that it's interactive, so I don't even try clicking in it. The previous widget at least looked like an actual editable input. Maybe use an unstyled (white/outlined) <input>, or switch back to the original widget for now?

@jankeromnes I have updated the layout for the widget doe this look ok ?

![image](https://user-images.githubusercontent.com/46004116/86315786-8990fd00-bc44-11ea-801b-8e13d8335a16.png)

nisarhassan12 commented:
@jankeromnes I think I have finished all the remainig items apart from a few see the comments above Thanks.

jankeromnes commented:
Thanks a lot! Merging as is (a few things left to fix, but this PR is getting too big)
